### PR TITLE
Slow Down Robots for Stop Command

### LINKED ISSUE #1882: Slow down robots when responding to stop command
### Description of the task
According to [SSL Rules](https://robocup-ssl.github.io/ssl-rules/sslrules.html#_stopping_the_game), we need to slow down robots to 1.5m/s when the stop command is issued. Currently, we don't support that behaviour. We used to have slow flag on the MoveAction/MoveIntent, but that may not be the best approach

### Acceptance criteria

- [ ] Slow down robots when responding to stop command
- [ ] add tests to make sure speed is slow

### Blocked By
https://github.com/UBC-Thunderbots/Software/issues/1959

### PR DESCRIPTION
**Pull request by BrianS99**
### Description

Adds robot stop-play behavior that slows down robots to 1.5 m/s when stop play is running, according to RoboCup SSL rules.

- Edited `stop_play.cpp` to pass Max Allowed Speed flag to Move Tactics and Goalie Tactics

### Testing Done
Used Tara's validation functions and stop-play test which was merged earlier, `stop_play_test.cpp`

### Resolved Issues
Resolves #1882

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/ai/hl/stp/play/stop_play_test.cpp [hunk 1]
```diff
@@ -26,7 +26,7 @@ class StopPlayTest : public SimulatedPlayTestFixture
                 if (world_ptr->getMostRecentTimestamp() >= Timestamp::fromSeconds(2))
                 {
                     // TODO: (#1882) implement robots slow down for stop play
```
Comment (akhilveeraghanta) [inline src/software/ai/hl/stp/play/stop_play_test.cpp]: nit: can remove this TODO here

**Review by akhilveeraghanta — COMMENTED** [review]
lgtm! nice job :tada: just 1 comment from me

**Review by jonathanlew — COMMENTED** [review]
LGTM pending other reviewers' comments

File: src/software/ai/hl/stp/play/stop_play_test.cpp (hunk 1 above)
Comment (BrianS99) [reply src/software/ai/hl/stp/play/stop_play_test.cpp]: Got it, thank you

**Review by akhilveeraghanta — DISMISSED** [review]
lgtm :fire: pending merge conflict fix

**Review by akhilveeraghanta — APPROVED** [review]

**Review by PhilD71 — COMMENTED** [review]
lgtm

**Review by PhilD71 — APPROVED** [review]

**Pull request merged**